cfme/middleware/provider/__init__.py

```diff
@@ -368,15 +369,27 @@ def add_jdbc_driver(self, filename, driver_name, module_name, driver_class,
         """
         self.load_details(refresh=True)
         jdbc_btn("Add JDBC Driver")
-        fill(jdbc_driver_form,
-            {
-                "file_select": filename,
-                "jdbc_driver_name": driver_name,
-                "jdbc_module_name": module_name,
-                "jdbc_driver_class": driver_class,
-                "major_version": major_version,
-                "minor_version": minor_version
-            })
+        if self.appliance.version >= '5.8':
```

Comment:
Because fill won't fill a None values, you can just construct the dict giving the ```driver_xa_datasource_class``` a None value if it doesn't exist. Then you can avoid version picking.